op-node/sources/receipts.go

```diff
@@ -82,13 +91,387 @@ func makeReceiptRequest(txHash common.Hash) (*types.Receipt, rpc.BatchElem) {
 	}
 }
 
-// NewReceiptsFetcher creates a receipt fetcher that can iteratively fetch the receipts matching the given txs.
-func NewReceiptsFetcher(block eth.BlockID, receiptHash common.Hash, txHashes []common.Hash, getBatch BatchCallContextFn, batchSize int) eth.ReceiptsFetcher {
-	return NewIterativeBatchCall[common.Hash, *types.Receipt, types.Receipts](
-		txHashes,
-		makeReceiptRequest,
-		makeReceiptsFn(block, receiptHash),
-		getBatch,
-		batchSize,
-	)
+// Cost break-down sources:
+// Alchemy: https://docs.alchemy.com/reference/compute-units
+// QuickNode: https://www.quicknode.com/docs/ethereum/api_credits
+// Infura: no pricing table available.
+//
+// Receipts are encoded the same everywhere:
+//
+//     blockHash, blockNumber, transactionIndex, transactionHash, from, to, cumulativeGasUsed, gasUsed,
+//     contractAddress, logs, logsBloom, status, effectiveGasPrice, type.
+//
+// Note that Alchemy/Geth still have a "root" field for legacy reasons,
+// but ethereum does not compute state-roots per tx anymore, so quicknode and others do not serve this data.
+
+// RPCProviderKind identifies an RPC provider, used to hint at the optimal receipt fetching approach.
+type RPCProviderKind string
+
+const (
+	RPCKindAlchemy    RPCProviderKind = "alchemy"
+	RPCKindQuickNode  RPCProviderKind = "quicknode"
+	RPCKindInfura     RPCProviderKind = "infura"
+	RPCKindParity     RPCProviderKind = "parity"
+	RPCKindNethermind RPCProviderKind = "nethermind"
+	RPCKindDebugGeth  RPCProviderKind = "debug_geth"
+	RPCKindErigon     RPCProviderKind = "erigon"
+	RPCKindBasic      RPCProviderKind = "basic" // try only the standard most basic receipt fetching
+	RPCKindAny        RPCProviderKind = "any"   // try any method available
+)
+
+func (kind RPCProviderKind) String() string {
+	return string(kind)
+}
+
```

Comment:
You should make a type for `RPCKinds` and make it a stringer so that you can render it in the CLI `--help` command

Reply:
implemented a util instead, I imagine we have more enums we may want to show in flags.